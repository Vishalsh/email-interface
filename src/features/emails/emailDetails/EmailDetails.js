import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Badge from 'components/badge/Badge';
import classes from './EmailDetails.module.scss';

export const EmailDetails = (props) => {
  const {
    email
  } = props;

  return (
    <section className={classes.emailDetails}>
      <div className="row middle-sm">
        <h1>{email.subject}</h1>
        <div>
          <Badge className={email.category} label={email.category}/>
        </div>
      </div>

      <div className={`${classes.detailRow} row`}>
        <div className="col-sm-6">
          <div>
            <span className={classes.label}>From: </span>
            <span>{`${email.sender.name} <${email.sender.email}>`}</span>
          </div>
          <div>
            <span className={classes.label}>To: </span>
            <span>{email.to}</span>
          </div>
        </div>
        <div className="col-sm-6 end-sm">
          <span>{email.dateTime}</span>
        </div>
      </div>

      <p>{email.body}</p>
    </section>
  );
};

EmailDetails.propTypes = {
  email: PropTypes.object.isRequired
};

const mapStateToProps = (store, ownProps) => ({
  email: store.emails[ownProps.match.params.id]
});

export default connect(mapStateToProps)(EmailDetails);
