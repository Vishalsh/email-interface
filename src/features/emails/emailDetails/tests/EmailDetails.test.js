import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { EmailDetails } from '../EmailDetails';

describe('Emails', () => {
  const props = {
    email: {
      id: 1,
      subject: 'subject 1',
      category: 'documents',
      sender: {
        name: 'MS Dhoni',
        email: 'msDhoni@bcci.com'
      },
      dateTime: '25 Dec',
      body: 'Lorem ipsum dolor sit amet.'
    }
  };
  const component = shallow(<EmailDetails {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});