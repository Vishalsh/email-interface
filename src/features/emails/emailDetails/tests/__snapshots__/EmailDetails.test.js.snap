// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emails should render the component 1`] = `
<section
  className="emailDetails"
>
  <div
    className="row middle-sm"
  >
    <h1>
      subject 1
    </h1>
    <div>
      <Badge
        className="documents"
        label="documents"
      />
    </div>
  </div>
  <div
    className="detailRow row"
  >
    <div
      className="col-sm-6"
    >
      <div>
        <span
          className="label"
        >
          From: 
        </span>
        <span>
          MS Dhoni &lt;msDhoni@bcci.com&gt;
        </span>
      </div>
      <div>
        <span
          className="label"
        >
          To: 
        </span>
        <span />
      </div>
    </div>
    <div
      className="col-sm-6 end-sm"
    >
      <span>
        25 Dec
      </span>
    </div>
  </div>
  <p>
    Lorem ipsum dolor sit amet.
  </p>
</section>
`;
